ui/v2.5/src/components/Tagger/utils.ts

```diff
@@ -92,55 +93,77 @@ const handleSpecialStrings = (input: string): string => {
   return output.replace(/-/g, " ");
 };
 
+function parseDate(input: string): string {
+  return input
+    .replace(/\./g, " ")
+    .replace(/ {2,}/g, " ")
+    .trim();
+}
+
 export function prepareQueryString(
   scene: Partial<GQL.SlimSceneDataFragment>,
   paths: string[],
   filename: string,
   mode: ParseMode,
   blacklist: string[]
 ) {
-  const regexs = blacklist
-    .map((b) => {
-      try {
-        return new RegExp(b, "gi");
-      } catch {
-        // ignore
-        return null;
-      }
-    })
-    .filter((r) => r !== null) as RegExp[];
+  let str = "";
 
   if ((mode === "auto" && scene.date && scene.studio) || mode === "metadata") {
-    let str = [
+    str = [
       scene.date,
       scene.studio?.name ?? "",
       (scene?.performers ?? []).map((p) => p.name).join(" "),
       scene?.title ? scene.title.replace(/[^a-zA-Z0-9 ]+/g, "") : "",
     ]
       .filter((s) => s !== "")
       .join(" ");
-    regexs.forEach((re) => {
-      str = str.replace(re, " ");
-    });
-    return str;
-  }
-  let s = "";
-
-  if (mode === "auto" || mode === "filename") {
-    s = filename;
-  } else if (mode === "path") {
-    s = [...paths, filename].join(" ");
-  } else if (mode === "dir" && paths.length) {
-    s = paths[paths.length - 1];
+  } else {
+    if (mode === "auto" || mode === "filename") {
+      str = filename;
+    } else if (mode === "path") {
+      str = [...paths, filename].join(" ");
+    } else if (mode === "dir" && paths.length) {
+      str = paths[paths.length - 1];
+    }
   }
 
-  regexs.forEach((re) => {
-    s = s.replace(re, " ");
+  const regexReplacements = new Map<string, string>();
+  const regexs = blacklist
+    .map((entry) => {
+      let [pattern, replacement] = entry.split("||"); // Extract regex and replacement
+
+      try {
+        const compiledRegex = new RegExp(pattern, "gi");
+
+        if (replacement  !== undefined ){
+          regexReplacements.set(compiledRegex.source, replacement); // Store replacement
+        }
+
+        return compiledRegex;
+      } catch {
+        return null; // Ignore invalid regex patterns
+      }
+    })
+    .filter((r) => r !== null) as RegExp[];
```

Comment:
It's not ideal to use `map` and then mutate stuff outside it. Instead, I would return a tuple (`return [compiledRegex, replacement]`). Then, these pairs of values can be handled in the code below. I would also move this into a separate function.